### ISSUE TITLE
Expose callback in XAxis to adjust tick values

### ISSUE THREAD

**Issue by 7imon7ays** [uber/react-vis#261]
There are at least two cases when I need to adjust the number of tick values shown on my X axis:
- When the window is resized.
- When the range covered by the X axis grows or shrinks.

These events happen in different places for most React apps so there's no elegant way to handle them.

A nice addition to the `XAxis` interface would be:
```
render() {
  return <XAxis tickValues={ [1, 2, 3] } onChange={ this.adjustTickValues }/>
}

adjustTickValues(tickValues, oldWidth, newWidth) {
  // Naive approach. Ideally the new size of `tickValues` is taken into account too.
  const adjustmentFactor = newWidth > 700 ? 1 : newWidth > 300 ? 2 : 4;
  return tickValues.filter((_, idx) => idx % adjustmentFactor === 0);
}
```

It also would be nice if the `onChange` callback somehow allowed me to overwrite `tickFormat` so that I can change the tick value precision based on how much of a range the X axis covers.

Even nicer if I could pass a boolean to get all this for free 🙌

**Comment by mcnuttandrew** [comment]
Hey @7imon7ays 

Generally this behavior is already built into react-vis. When things change the axes get updated. However, if you are specifically setting the tick values then you are saying that you know better than react-vis about how to lay things (which you most likely do!), but in exchange for that specificity you need to keep more track of your data. If you want more ganular control over the exact format of your ticks, but still want us to handle the layout of the ticks for you, you should be using tickFormat instead.

TLDR: if you want to exactly specify your ticks, you need to be in charge of updating them.
